### ISSUE TITLE
Setup fails with SSL: WRONG_VERSION_NUMBER

### ISSUE THREAD

**Issue by sabeechen** [freeipa/freeipa-container#442]
Initial setup of the container is failing with the the following error in ipaserver-install.log after starting the CA server:
```
INFO: Creating /etc/pki/pki-tomcat/Catalina/localhost/ca.xml
INFO: Starting PKI server
DEBUG: Command: systemctl start pki-tomcatd@pki-tomcat.service
INFO: Waiting for PKI server to start
Exception: Server unreachable due to SSL error: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:897)
  File "/usr/lib/python3.6/site-packages/pki/server/pkispawn.py", line 575, in main
    scriptlet.spawn(deployer)
  File "/usr/lib/python3.6/site-packages/pki/server/deployment/scriptlets/configuration.py", line 702, in spawn
    timeout=deployer.request_timeout)
  File "/usr/lib/python3.6/site-packages/pki/server/__init__.py", line 354, in start
    raise Exception('Server unreachable due to SSL error: %s' % reason) from e
````

I've tried running fedora35, fedora35, fedora33, and centos8 images all with the same error.  The host operating system is centos8.

My docker-compose for the container is built from the documentation (ips and hostnames changed for privacy):
```
version: '2'
services:
  free-ipa:
    image: freeipa/freeipa-server:rocky-8
    container_name: free-ipa
    tty: true
    stdin_open: true
    hostname: ipa1
    domainname: example.com
    dns:
      - 192.168.0.1
    extra_hosts:
      - "ipa1.example.com:192.168.0.9"
    environment:
      - IPA_SERVER_HOSTNAME=ipa1.example.com
      - IPA_SERVER_IP=192.168.0.9
    volumes:
      - /config/containers/free-ipa/data:/data
      - /sys/fs/cgroup:/sys/fs/cgroup:ro
    cap_add:
      - SYS_TIME
    sysctls:
      - "net.ipv6.conf.all.disable_ipv6=0"
    ports:
      # ... a bunch of ports ...

```

I was able to find [this forum post](https://forums.centos.org/viewtopic.php?f=54&t=76238) which seems to suggest the issue is with a specific java version, but I find it hard to believe I'd be the only one having this problem is that were the case and I couldn't find any other issues referencing this error.

**Comment by adelton** [comment]
I'd assume the WRONG_VERSION_NUMBER is a red herring and the issue might be in the general networking/hostname resolution.

Could you test the installation with plain `docker run` instead of `docker-compose`?

**Comment by sabeechen** [comment]
I had the same result when I rewrote it to use ```docker run```, but you pointed me in the right direction.  I had changed network settings on the host without restarting docker and as a result the container's internal network wasn't working correctly and was unable to contact the DNS server from inside the container (eh ```host ipa1.example.com``` would timeout).  After restarting docker and ensuring the container was able to resolve names correctly (especially its own name) I was able to get the container running correctly.  I hadn't suspected that WRONG_SSL_VERSION could be cause by networking/DNS issues.

Thank you for your help!
